### ISSUE TITLE
Pub/Sub Template Sample fails with CNF

### ISSUE THREAD

**Issue by meltsufin** [spring-cloud/spring-cloud-gcp#751]
```
java.lang.NoClassDefFoundError: org/springframework/integration/support/json/JacksonJsonUtils
```

The reason is that `JacksonJsonUtils` seems to be required for the autoconfiguration, but the dependency on `spring-integration-core` is `test` scoped.

**Comment by artembilan** [comment]
That's why I suggested before do not depend on Spring Integration there at all.
We can configure something though, if SI is in the classpath. but if not - fallback to something else.

**Comment by meltsufin** [comment]
Fixed by #759.
